Toggl.Storage.Realm/Queries/ProcessPushResultQuery.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Linq;
+using System.Reactive;
+using Toggl.Networking.Sync.Push;
+using Toggl.Shared;
+using Toggl.Shared.Models;
+using Toggl.Storage.Queries;
+using Toggl.Storage.Realm.Models;
+
+namespace Toggl.Storage.Realm.Queries
+{
+    internal class ProcessPushResultQuery : IQuery<Unit>
+    {
+        private readonly IResponse response;
+        private readonly Func<Realms.Realm> realmProvider;
+
+        public ProcessPushResultQuery(Func<Realms.Realm> realmProvider, IResponse response)
+        {
+            Ensure.Argument.IsNotNull(realmProvider, nameof(response));
+            Ensure.Argument.IsNotNull(response, nameof(response));
+
+            this.realmProvider = realmProvider;
+            this.response = response;
+        }
+
+        public Unit Execute()
+        {
+            var realm = realmProvider();
+            using (var tx = realm.BeginWrite())
+            {
+                processSingletonResult<IPreferences, RealmPreferences>(response.Preferences, realm);
+                processSingletonResult<IUser, RealmUser>(response.User, realm);
+
+                foreach (var res in response.Workspaces)
+                    processResult<IWorkspace, RealmWorkspace>(res, realm);
+
+                foreach (var res in response.Tags)
+                    processResult<ITag, RealmTag>(res, realm);
+
+                foreach (var res in response.Clients)
+                    processResult<IClient, RealmClient>(res, realm);
+
+                foreach (var res in response.Tasks)
+                    processResult<ITask, RealmTask>(res, realm);
+
+                foreach (var res in response.Projects)
+                    processResult<IProject, RealmProject>(res, realm);
+
+                foreach (var res in response.TimeEntries)
+                    processResult<ITimeEntry, RealmTimeEntry>(res, realm);
+
+                tx.Commit();
```

Comment:
```suggestion
                transaction.Commit();
```

Don't be that lazy 😊